modules/rpc/yarpc.go

```diff
@@ -62,13 +67,30 @@ var (
 
 type registerServiceFunc func(module *YARPCModule)
 
+type transports struct {
+	inbounds []transport.Inbound
+}
+
 type yarpcConfig struct {
 	modules.ModuleConfig
-	Bind                    string
-	AdvertiseName           string `yaml:"advertiseName"`
+
 	inboundMiddleware       []middleware.UnaryInbound
 	onewayInboundMiddleware []middleware.OnewayInbound
-	inbounds                []transport.Inbound
+
+	transports transports
+	Inbounds   []Inbound
+}
+
+// Inbound is a union that configures how to configure a single inbound.
+type Inbound struct {
+	TChannel *Port
+	HTTP     *Port
+}
+
+// Port is a struct that have a required port for tchannel/http transports.
+// TODO(alsam) make it optional
+type Port struct {
```

Comment:
It is not a type that I am after, but structure: to show port for both tchannel and http:
```
Inbounds:
- http
   8080
- tchannel
   8181
```